api/src/main/java/com/velocitypowered/api/util/GameProfile.java

```diff
@@ -8,24 +8,36 @@
 /**
  * Represents a Mojang game profile. This class is immutable.
  */
-public final class GameProfile {
+public final class GameProfile implements Identifiable {
 
-  private final String id;
-  private final String name;
+  private final UUID id;
```

Comment:
To negate the need for a custom `GameProfileSerializer`:

* Keep `id` as `String`
* Add a new `UUID` field (say `javaUuid`) with `transient` keyword.

Reply:
The problem is that it would allow to create an invalid `GameProfile` object when the `id` is not a valid undashed uuid which can cause an exception when calling `GameProfile#idAsUuid`. Do you think such objects should be allowed ? It does not only affect gson, it was also possible by calling the constructor with an invalid parameter.

Reply:
If that's a concern I would probably validate the ID. This can be done trivially.

Reply:
Actually, you are correct - this just hasn't come up in practice. I'm fine with keeping the `GameProfileSerializer`.

Reply:
I saw a message saying that you're fine with keeping the `GameProfileSerializer` but this message is no longer here. Do you still want to keep it or not ?

Reply:
We can keep it.